core/src/stages/compute_ik.cpp

```diff
@@ -353,14 +360,25 @@ void ComputeIK::compute() {
 	                 &ik_solutions](robot_state::RobotState* state, const robot_model::JointModelGroup* jmg,
 	                                const double* joint_positions) {
 		for (const auto& sol : ik_solutions) {
-			if (jmg->distance(joint_positions, sol.data()) < min_solution_distance)
+			if (jmg->distance(joint_positions, sol.joint_positions.data()) < min_solution_distance)
 				return false;  // too close to already found solution
 		}
 		state->setJointGroupPositions(jmg, joint_positions);
 		ik_solutions.emplace_back();
-		state->copyJointGroupPositions(jmg, ik_solutions.back());
+		auto& solution{ ik_solutions.back() };
+		state->copyJointGroupPositions(jmg, solution.joint_positions);
+		collision_detection::CollisionRequest req;
+		collision_detection::CollisionResult res;
+		req.contacts = true;
+		req.max_contacts = 1;
+		scene->checkCollision(req, res, *state);
+		solution.feasible = ignore_collisions || !res.collision;
```

Comment:
When `ignore_collisions` is `true`, the collision checking should be skipped, shouldn't it?

Reply:
We discussed tha when writing the patch and I am in favor of *always* doing collisions checks here to provide the feedback even if the solution is still marked as feasible. I notice now that checking was indeed skipped before in this case (as obviously no feedback was generated from it), but I would still change it in favor of the additional feedback.

If you insist it's better to retain the current behavior here in favor of performance we can do it as well, but `ignore_collisions=true` was always more of a debugging flag for me.

Reply:
You are right. The default case is to perform collision checking anyway. I just wanted to point out the difference in behavior.